test/plugin.test.js

```diff
@@ -0,0 +1,349 @@
+import document from 'global/document';
+
+import QUnit from 'qunit';
+import sinon from 'sinon';
+import videojs from 'video.js';
+
+import plugin from '../src/plugin';
+
+const Player = videojs.getComponent('Player');
+
+QUnit.test('the environment is sane', function(assert) {
+  assert.strictEqual(typeof Array.isArray, 'function', 'es5 exists');
+  assert.strictEqual(typeof sinon, 'object', 'sinon exists');
+  assert.strictEqual(typeof videojs, 'function', 'videojs exists');
+  assert.strictEqual(typeof plugin, 'function', 'plugin is a function');
+});
+
+QUnit.module('videojs-errors', {
+
+  beforeEach() {
+
+    // Mock the environment's timers because certain things - particularly
+    // player readiness - are asynchronous in video.js 5.
+    this.clock = sinon.useFakeTimers();
+    this.fixture = document.getElementById('qunit-fixture');
+    this.video = document.createElement('video');
+    this.fixture.appendChild(this.video);
+    this.player = videojs(this.video);
+
+    this.player.buffered = function() {
+      return videojs.createTimeRange(0, 0);
+    };
+    this.player.paused = function() {
+      return false;
+    };
+    this.player.pause = function() {
+      return false;
+    };
+
+    // initialize the plugin with the default options
+    this.player.errors();
+
+    // Tick forward so the player is ready.
+    this.clock.tick(1);
+  },
+
+  afterEach() {
+    this.player.dispose();
+    this.clock.restore();
+  }
+});
+
+QUnit.test('registers itself with video.js', function(assert) {
+  assert.expect(2);
+
+  assert.strictEqual(
+    Player.prototype.errors,
+    plugin,
+    'videojs-errors plugin was registered'
+  );
+
+  this.player.errors();
+
+  // Tick the clock forward enough to trigger the player to be "ready".
+  this.clock.tick(1);
+
+  assert.ok(
+    this.player.hasClass('vjs-errors'),
+    'the plugin adds a class to the player'
+  );
+});
+
+QUnit.test('play() without a src is an error', function(assert) {
+  let errors = 0;
+
+  this.player.on('error', function() {
+    errors++;
```

Comment:
I've been replacing these counters with `sinon.spy()` functions in most places, but that's probably not necessary. Just throwing that out there as an alternative to this practice. I don't personally have strong feelings on it.